WordPress-VIP/ruleset.xml

```diff
@@ -1,127 +0,0 @@
-<?xml version="1.0"?>
```

Comment:
Can you ellaborate on this? e.g. would a more verbose version of this suffice?

```
<?php
trigger_error('deprecated');
```

Or an echo? Or is there a PHPCS API needed?

Reply:
Eh... this is an XML file ? Don't know why you'd think you could use PHP here.

> Or is there a PHPCS API needed?

I have no clue what you mean by this.

Anyway, there are two approaches:
1. Clear out the ruleset + add a bootstrap file which will fail builds with a deprecation notice for the ruleset.

    In that case, just make it an empty ruleset:

    ```xml
    <?xml version="1.0"?>
    <ruleset name="WordPress VIP">
    	<description>The WordPress.com VIP Coding Standards have been moved.</description>
    </ruleset>
    ```

    You can even leave out the description, but I left it in as a form of documentation if people open it because they find it strange things aren't working anymore.

2. Leave the ruleset as it is. Add a bootstrap file with a deprecation notice which won't exit with a fail code.
That way things will still work for people using the `WordPress-VIP` ruleset, but if they pay attention, they will get notified of the deprecation and can't be surprised by it anymore by the time we remove the ruleset completely.

Reply:
I mean in the bootstrap PHP file, I didn't realised you meant literally just an empty ruleset.xml with minimal tags

Reply:
> I mean in the bootstrap PHP file

Ah, well, there you would need to check what PHPCS has registered.
Something along the lines of:
* If the `WordPress-VIP` ruleset was the main ruleset being included, I'd suggest echo-ing out a deprecation message and doing and `exit(1)`.
* If the `WordPress-VIP` ruleset was registered as part of a larger ruleset, you may just want to throw the deprecation message, but not throw an exit.
* If no reference to `WordPress-VIP` was found, do nothing and let PHPCS continue.